model/src/appdata_hexadecimal.rs

```diff
@@ -0,0 +1,51 @@
+use serde::{de, Deserializer, Serializer};
+use std::fmt;
+
+pub fn serialize<S>(value: &[u8; 32], serializer: S) -> Result<S::Ok, S::Error>
+where
```

Comment:
With the next rust version we will get const generics with which we might be able to implement a single serialize thing for all the places we want to encode bytes as hex (h160, signature, appdata).